### PR TITLE
Enable http2 server by default

### PR DESCRIPTION
**Pull request by jgraham**

### CODE REVIEW

**Comment by gsnedders** [comment]
Can you add an infrastructure test that at least does *something* with it?

File: infrastructure/server/http2-context.sub.h2.any.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+test(() => {
+  assert_equals(location.protocol, "https:");
+}, "Use of .h2. file name flag implies HTTPS scheme");
+
+test(() => {
+  assert_equals(location.port, "{{ports[h2][0]}}");
+}, "Use of .h2. file name flag implies correct port");
+
+done();
```
Comment (Ms2ger) [inline infrastructure/server/http2-context.sub.h2.any.js]: Don't think you need this at this point.

File: docs/writing-tests/file-names.md [hunk 2]
```diff
@@ -29,6 +29,9 @@ themselves precede any test type flag, but are otherwise unordered.
 `.https`
  : Indicates that a test is loaded over HTTPS.
 
+ `.h2`
```
Comment (gsnedders) [inline docs/writing-tests/file-names.md]: Now this is adding a new test type which seems like a more major addition than just enabling by default and perhaps worthy of an RFC; when I said write tests I was more expecting something just loading something cross-origin and checking the server is responding (and given some of the other changes in the PR now, I wonder if it wasn't).

Comment (jgraham) [reply docs/writing-tests/file-names.md]: The RFC is now merged.

File: tools/manifest/sourcefile.py [hunk 3]
```diff
@@ -156,6 +156,8 @@ def global_variant_url(url, suffix):
     # the form .https.any.js
     if ".https." in url and suffix.startswith(".https."):
         url = url.replace(".https.", ".")
+    elif ".h2." in url and suffix.startswith(".h2."):
+        url = url.replace(".h2.", ".")
```
Comment (gsnedders) [inline tools/manifest/sourcefile.py]: does this need a manifest bump? or does it not because there are no such files currently?

Comment (jgraham) [reply tools/manifest/sourcefile.py]: There are no such files currently.

**Review by gsnedders — APPROVED** [review]
Pretty sure I meant to approve this after @jgraham's comment at https://github.com/web-platform-tests/wpt/pull/20620#discussion_r362916595

**Pull request merged**